### ISSUE TITLE
magento2:generate:view  generates broken layout (m2.2)

### ISSUE THREAD

**Issue by eralaz** [astorm/pestle#341]
Generates layout like

     <page>
        <referenceBlock>
       ...

when from m2.2 <body> should be between them, like this

    <page>
      <body>
        <referenceBlock>
    ...

**Comment by astorm** [comment]
@larylaz Thank you for the feedback.  However, it's not clear to me what you're saying.  Your post looks like this to me 

    Generates layout like


    ...

    when from m2.2 should be between them, like this
    ... 

I'm not sure what the ellipses represent.  Coud you try describing your problem in a different way so we can understand what you mean?

**Comment by eralaz** [comment]
Hi)
briefly the issue is magento2:generate:view generates broken layout

**Comment by eralaz** [comment]
layout broken due to it missed <body> tag within the layout xml file

**Comment by eralaz** [comment]
I see, the snippet has been masked. So, you've got part of the description.

**Comment by astorm** [comment]
@larylaz GitHub lets you post  code samples by indenting this by four characters.  If you can include the problematic output we can take a look at the problem.

Also -- can you tell me what "broken" means?  Does it cause an error?  Just not render? What's the error?  Hoe does it render incorrectly? What version of Magento, etc.  

We'd love to help, but without this information we're not sure what to do.

**Comment by astorm** [comment]
@larylaz did you solve your problem?

**Comment by eralaz** [comment]
yep, just manually fix the storefront page generated layout. And I'm on Magento version of 2.2.2.
It seems like 2.1 had a bid different page layout structure.

**Comment by astorm** [comment]
@larylaz We're glad to hear the problem's fixed.  We'd still love it if you could share

**Comment by astorm** [comment]
@larylaz I was able to edit your oiginal comment to get at the actual problem you were having.  What command did you run to create a layout where the lack of a  `<body/>` makes Magento 2 freak out?  Our tests didn't show this happening.  We're not doubting you, but without further details we don't know what to fix for you.

**Comment by eralaz** [comment]
the command I was used was "pestle magento2:generate:view"

**Comment by astorm** [comment]
@larylaz with what sort of arguments (you can substitute different strings if the names are sensitive).  What sort of errors did the generated file produce?

**Comment by eralaz** [comment]
there were no args pure command and there were no errors on the cli screen.

**Comment by astorm** [comment]
@larylaz It's not clear what you mean by a "pure command".  

When you run that command pestle asks you for arguments

```
$ pestle_dev magento2:generate:view
Which Module? (Pulsestorm_HelloGenerate)] 
Which Area? (frontend)] 
Which Handle? (pulsestorm_hellogenerate_index_index)] 
Block Name? (Main)] 
Template File? (content.phtml)] 
```

What values did you use for those arguments?

Also 

> no errors on the cli screen

We understand that there were no errors in the command line screen.  However, you presumably opened this bug because the layout created errors when you tried to use it in your Magento system.  What error was that?

Or were there no errors in Magento, and this is a feature request for the command to generate the layout XML files with `<body/>` tags?

Answers to these questions will help us ensure we're adding both the features you need, as well as the feature other pestle developers will need.

**Comment by eralaz** [comment]
Yep!
This is a feature request for the command to generate the layout XML files with <body/> tags. That's the main point.
